### ISSUE TITLE
Fuzz errors in 1.0.0-rc.4 and master (8/21)

### ISSUE THREAD

**Issue by phet-steele** [phetsims/charges-and-fields#63]
This failed assertion was found in master, NOT 1.0.0-rc.4:

```
Error: Assertion failed: the chargedParticles array must contain at least one element

window.assertions.assertFunction (assert.js:22)
inherit.getClosestChargedParticlePosition (ElectricPotenti…0186915843:324)
inherit.getIsLinePresent (ElectricPotenti…40186915843:77)
ElectricPotentialLine (ElectricPotenti…40186915843:50)
inherit.addElectricPotentialLine (ChargesAndField…0186915843:580)
ElectricPotentialSensorNode.PencilButton.listener (ElectricPotenti…0186915843:102)
(anonymous function) (PushButtonModel…0186915843:126)
inherit.fire (PushButtonModel…0186915843:125)
(anonymous function) (PushButtonModel…40186915843:72)
onValueObserver (Property.js?bus…0186915843:276)
inherit._notifyObservers (Property.js?bus…0186915843:112)
inherit._setAndNotifyObservers (Property.js?bus…40186915843:99)
inherit.set (Property.js?bus…40186915843:72)
Object.defineProperty.set (PropertySet.js?…0186915843:136)
ButtonListener.DownUpListener.call.up (ButtonListener.…40186915843:49)
inherit.buttonUp (DownUpListener.…0186915843:121)
DownUpListener.downListener.up (DownUpListener.…40186915843:55)
inherit.dispatchToPointer (Input.js?bust=1…0186915843:798)
inherit.dispatchEvent (Input.js?bust=1…0186915843:766)
inherit.upEvent (Input.js?bust=1…0186915843:631)
inherit.mouseUp (Input.js?bust=1…0186915843:313)
extend.fuzzMouseEvents (Display.js?bust…0186915843:860)
animationLoop (Sim.js?bust=1440186915843:687)
```

and this error was found in 1.0.0-rc.4, NOT master (unfortunately this stack trace may be useless).

```
Uncaught TypeError: Cannot read property 'distance' of undefined

n.getIsLinePresent (charges-and-fie…zzMouse&ea:521)
e (charges-and-fie…zzMouse&ea:521)
s.addElectricPotentialLine (charges-and-fie…zzMouse&ea:521)
d.listener (charges-and-fie…zzMouse&ea:532)
(anonymous function) (charges-and-fie…zzMouse&ea:531)
r.fire (charges-and-fie…zzMouse&ea:531)
(anonymous function) (charges-and-fie…zzMouse&ea:531)
i.onValue.i (charges-and-fie…zzMouse&ea:516)
i._notifyObservers (charges-and-fie…zzMouse&ea:516)
i._setAndNotifyObservers (charges-and-fie…zzMouse&ea:516)
i.set (charges-and-fie…zzMouse&ea:516)
a.addGetterAndSetter.Object.defineProperty.set (charges-and-fie…zzMouse&ea:516)
i.call.up (charges-and-fie…zzMouse&ea:531)
e.buttonUp (charges-and-fie…zzMouse&ea:524)
i.DownUpListener.downListener.up (charges-and-fie…zzMouse&ea:524)
e.dispatchToPointer (charges-and-fie…zzMouse&ea:524)
e.dispatchEvent (charges-and-fie…zzMouse&ea:524)
e.upEvent (charges-and-fie…zzMouse&ea:524)
e.mouseUp (charges-and-fie…zzMouse&ea:523)
i.fuzzMouseEvents (charges-and-fie…zzMouse&ea:538)
t (charges-and-fie…zzMouse&ea:539)
```

**Comment by samreid** [comment]
I saw this in the context of https://github.com/phetsims/tasks/issues/310 and wanted to report on it a bit more:

The charges-and-fields issue is:

charges-and-fields
Uncaught TypeError: Cannot read property 'distance' of undefined
TypeError: Cannot read property 'distance' of undefined
    at e.n.getIsLinePresent (http://localhost:8080/charges-and-fields/build/charges-and-fields_en.html?ea&postMessageOnLoad&postMessageOnError&fuzzMouse=100:529:13824)

The code in question is:

``` js
    /**
     * Method that determines if an equipotential line is present based on
     * (1) are there charges on the board?, (2) is the initial point of the search far away from all the charges?
     * If both answers are yes, the function returns true;
     *
     * @public
     * @returns {boolean}
     */
    getIsLinePresent: function() {
      var isLinePresent; // {boolean}
      if ( !this.isPlayAreaChargedProperty.value ) {
        // if there are no charges, don't bother to find the electricPotential line
        isLinePresent = false;
      }
      else {
        var closestChargeDistance = this.getClosestChargedParticlePosition( this.position ).distance( this.position );
        var closestAllowedDistance = 0.03; // in model coordinates, should be less than the radius (in the view) of a charged particle

        // if the initial point for the electricPotential line search is too close to a charged particle, then
        // the equipotential line will not be visible
        // also see https://github.com/phetsims/charges-and-fields/issues/5
        isLinePresent = (closestChargeDistance > closestAllowedDistance);

      }
      return isLinePresent;
    }
```

So the problem is that `isPlayAreaCharged` is coming up true, but getClosestChargedParticlePosition is coming up undefined.

**Comment by samreid** [comment]
Changing this method in ElectricPotentialLine to read:

``` js
    getIsLinePresent: function() {
      var isLinePresent; // {boolean}
      if ( this.chargedParticles.length === 0 ) {
        // if there are no charges, don't bother to find the electricPotential line
        isLinePresent = false;
      }
```

fixes the probem:
![image](https://cloud.githubusercontent.com/assets/679486/9625150/de23c45a-5110-11e5-8524-f3ea7fea16c0.png)

This may be a fine fix for the fuzz test, but it also indicates that it is possible for isPlayAreaChargedProperty to get out of sync with this.chargedParticles.length.  @phetsims/bereaphysics are you available to evaluate this solution?  If you're busy, no worries.

**Comment by veillette** [comment]
I will look into this very shortly.

**Comment by jonathanolson** [comment]
Still an issue in the latest testing:

```
Error: Assertion failed:  the chargedParticles array must contain at least one element
    at window.assertions.assertFunction (http://192.168.0.2:8080/assert/js/assert.js:22:13)
    at ElectricPotentialLine.inherit.getClosestChargedParticlePosition (http://192.168.0.2:8080/charges-and-fields/js/charges-and-fields/model/ElectricPotentialLine.js?bust=1444430604814:324:17)
    at ElectricPotentialLine.inherit.getIsLinePresent (http://192.168.0.2:8080/charges-and-fields/js/charges-and-fields/model/ElectricPotentialLine.js?bust=1444430604814:77:42)
    at new ElectricPotentialLine (http://192.168.0.2:8080/charges-and-fields/js/charges-and-fields/model/ElectricPotentialLine.js?bust=1444430604814:50:31)
    at ChargesAndFieldsModel.inherit.addElectricPotentialLine (http://192.168.0.2:8080/charges-and-fields/js/charges-and-fields/model/ChargesAndFieldsModel.js?bust=1444430604814:580:37)
    at ElectricPotentialSensorNode.PencilButton.listener (http://192.168.0.2:8080/charges-and-fields/js/charges-and-fields/view/ElectricPotentialSensorNode.js?bust=1444430604814:102:9)
    at http://192.168.0.2:8080/sun/js/buttons/PushButtonModel.js?bust=1444430604814:126:9
    at Array.forEach (native)
    at PushButtonModel.inherit.fire (http://192.168.0.2:8080/sun/js/buttons/PushButtonModel.js?bust=1444430604814:125:12)
    at http://192.168.0.2:8080/sun/js/buttons/PushButtonModel.js?bust=1444430604814:72:25
```

**Comment by jessegreenberg** [comment]
The sim passes through test-server cleanly.  I let the mouseFuzz run for a few hours, never caught anything.

![charges-and-fields-clean](https://cloud.githubusercontent.com/assets/6396244/10795084/abec6018-7d5d-11e5-89f5-9ab6458398c8.PNG)

My guess is that this was fixed through issue #64 since that involved syncing the `isPlayAreaChargedProperty` with `this.chargedParticles.length`.

Closing.
